lib/ansible/plugins/connection/aws_ssm.py

```diff
@@ -0,0 +1,535 @@
+
+        # Wrap command in markers accordingly for the shell used
+        cmd = self._wrap_command(cmd, sudoable, mark_start, mark_end)
+
+        self._flush_stderr(session)
+
+        # Handle the back-end throttling
+        for c in cmd:
+            session.stdin.write(c.encode('utf-8'))
+            if self.is_windows:
+                time.sleep(15 / 1000.0)
+
+        # Read stdout between the markers
+        stdout = ''
+        begin = False
+        stop_time = int(round(time.time())) + self.get_option('timeout')
+        while session.poll() is None:
+            remaining = stop_time - int(round(time.time()))
+            if remaining < 1:
+                self._timeout = True
+                display.vvvv(u"EXEC timeout stdout: {0}".format(to_text(stdout)), host=self.host)
+                raise AnsibleConnectionFailure("SSM exec_command timeout on host: %s"
+                                               % self.instance_id)
+            if self._poll_stdout.poll(1000):
+                line = self._filter_ansi(self._stdout.readline())
+                display.vvvv(u"EXEC stdout line: {0}".format(to_text(line)), host=self.host)
+            else:
+                display.vvvv(u"EXEC remaining: {0}".format(remaining), host=self.host)
+                continue
+
+            if mark_start in line:
+                begin = True
+                if not line.startswith(mark_start):
+                    stdout = ''
+                continue
+            if begin:
+                if mark_end in line:
+                    for line in stdout.splitlines():
+                        display.vvvvv(u"POST_PROCESS: {0}".format(to_text(line)), host=self.host)
+                    returncode, stdout = self._post_process(stdout)
+                    break
+                else:
+                    stdout = stdout + line
+
+        stderr = self._flush_stderr(session)
+
+        return (returncode, stdout, stderr)
+
+    def _prepare_terminal(self):
+        ''' perform any one-time terminal settings '''
+
+        if not self.is_windows:
+            cmd = "stty -echo\n" + "PS1=''\n"
+            if PY3:
+                cmd = to_bytes(cmd)
+            self._session.stdin.write(cmd)
+
+    def _wrap_command(self, cmd, sudoable, mark_start, mark_end):
+        ''' wrap command so stdout and status can be extracted '''
+
+        if self.is_windows:
+            cmd = cmd + "; echo " + mark_start + " $? $LASTEXITCODE\necho " + mark_end + "\n"
+            time.sleep(1)
+        else:
+            if sudoable:
+                cmd = "sudo " + cmd
+            cmd = "echo " + mark_start + "\n" + cmd + "\necho $'\\n'$?\n" + "echo " + mark_end + "\n"
+
+        display.vvvv(u"_wrap_command: '{0}'".format(to_text(cmd)), host=self.host)
+        return cmd
+
+    def _post_process(self, stdout):
+        ''' extract command status and strip unwanted lines '''
+
+        if self.is_windows:
+            success = stdout.rfind('True')
```

Comment:
With the removal of `$?` this shouldn't be needed.

Reply:
@jborean93 , We have updated the code as you suggested in the commit https://github.com/ansible/ansible/pull/49652/commits/741623e3295a5f6af448da25855e1f13149b7676 and all the checks were passed. Can you please review and approve ?